packages/core/src/types.ts

```diff
@@ -74,6 +91,42 @@ export type Action<TContext, TEvent extends EventObject> =
   | ActionObject<TContext, TEvent>
   | ActionFunction<TContext, TEvent>;
 
+/**
+ * Extracts action objects that have no extra properties.
+ */
+type SimpleActionsFrom<T extends BaseActionObject> = ActionObject<
+  any,
+  any
+> extends T
+  ? T // If actions are unspecified, all action types are allowed (unsafe)
+  : T extends {
+      [K in keyof Omit<T, 'type'>]: never;
+    }
+  ? T
+  : never;
+
+export type BaseAction<
+  TContext,
+  TEvent extends EventObject,
+  TAction extends BaseActionObject
+> =
+  | SimpleActionsFrom<TAction>['type']
+  | TAction
+  | RaiseAction<any>
+  | SendAction<TContext, TEvent, any>
+  | AssignAction<TContext, TEvent>
+  | LogAction<TContext, TEvent>
+  | CancelAction
+  | StopAction<TContext, TEvent>
+  | ChooseAction<TContext, TEvent>
+  | ActionFunction<TContext, TEvent>;
+
+export type BaseActions<
```

Comment:
"Base" doesn't tell us much - what do you think about changing that to "Inline"? I'm not entirely sure if that was the intention but it seems to hold right now. And there might be some benefit in having `InlineAction` and `ImplementationAction` split anyway.

Reply:
The intention is for there to be an action object without `exec`; I typically use `Base*` as a type name to signify a type that other types build on. But this isn't for a specific type of action; it's for all actions.

Reply:
This would still, potentially, leave `BaseActionObject` untouched.

Reply:
Ah I thought you were talking about `BaseActionObject` - maybe naming this `PlainActions` would be better